src/Swashbuckle.AspNetCore.Cli/Program.cs

```diff
@@ -140,10 +140,24 @@ private static IServiceProvider GetServiceProvider(Assembly startupAssembly)
                 return webHost.Services;
             }
 
-            return WebHost.CreateDefaultBuilder()
-               .UseStartup(startupAssembly.GetName().Name)
-               .Build()
-               .Services;
+            try
+            {
+                return WebHost.CreateDefaultBuilder()
+                   .UseStartup(startupAssembly.GetName().Name)
+                   .Build()
+                   .Services;
+            }
+            catch
+            {
+                var serviceProvider = HostingApplication.GetServiceProvider(startupAssembly);
```

Comment:
@davidfowl you mentioned that the `HostFactoryResolver` is the de facto way to resolve entry point and service provider for apps that use `Microsoft.Extensions.Hosting`. With that in mind, shouldn't it be the default approach here rather than the fallback?

Reply:
I wanted to keep the existing compatibility.